builtin/v13/migration/market.go

```diff
@@ -309,10 +309,21 @@ func (m *marketMigrator) migrateProviderSectorsAndStatesFromScratch(ctx context.
 		return cid.Undef, cid.Undef, xerrors.Errorf("failed to load proposals array: %w", err)
 	}
 
+	proposalsSize := proposalsArr.Length()
+
 	var oldState market12.DealState
 	var newState market13.DealState
 
 	err = oldStateArray.ForEach(&oldState, func(i int64) error {
+		// Refresh proposals array periodically to avoid holding onto all ~10GB of memory
+		// throughout whole migration.
+		// This has limited impact on caching speedups because the access pattern is sequential.
+		if i%(int64(proposalsSize)/1000) == 0 {
```

Comment:
```suggestion
		if proposalsSize > 0 && i%(int64(proposalsSize)/1000) == 0 {
```

picked up @ https://app.circleci.com/pipelines/github/filecoin-project/lotus/33637/workflows/b64e4593-54cb-437f-bbdc-dbe756bd2280/jobs/1210829